sdk/appconfiguration/app-configuration/test/public/index.spec.ts

```diff
@@ -1140,4 +1155,269 @@ describe("AppConfigurationClient", () => {
       });
     });
   });
+
+  describe("FeatureFlag configuration setting", () => {
+    const clientFilters: (
+      | object
+      | FeatureFlagTargetingClientFilter
+      | FeatureFlagTimeWindowClientFilter
+      | FeatureFlagPercentageClientFilter
+    )[] = [
+      {
+        name: "Microsoft.TimeWindow",
+        parameters: {
+          start: "Wed, 01 May 2019 13:59:59 GMT",
+          end: "Mon, 01 July 2019 00:00:00 GMT"
+          // TODO: dates should accept `string | Date`
+          // Returning
+          //      - SDK would return Date for all the config-settings if parsable
+          //      - SDK would return strings for non parsable strings
+          // Passing as an argument
+          //      - If passed in a string, we'll pass it as is.
+          //      - If passed Date, we'll make it a string.
+        }
```

Comment:
Currently, dates are strings.. how is this proposal?

Reply:
@richardpark-msft @chradek

Reply:
Are the other languages going with just string?

I'm not against allowing customers to pass in a Date, but I don't love returning mixed types when it comes back from the service. I'm worried some customers (especially JS users) will see that it's a date and assume it's always a date, then get bit when it comes back as a string.

Reply:
I can leave it as string and probably add docs to show how they can get to that string from a given date while making adding a config setting.

Looking at the API Views of other languages... they are just leaving it as dictionaries for the "parameters"..
To confirm.. @pakrym @mssfang @seankane-msft Are you guys supporting the date type of your languages or just leaving it as is?

Reply:
I believe this was meant to be a point taken back to the service team to ask if they can standardize on a format that's more common, otherwise parsing it would be difficult in all languages. (I believe @pakrym was taking this as an action item but I could be misremembering).

So `string` might be the best type for now but it'd be nice (before the preview ends) to resolve this and make it so the expected type could be a `Date`. If we can actually support parsing this date on all platforms I'd be more than happy to just have it be Date as a type since the primary use case of this would be to compare against the current date/time and see if we've entered the time window for this feature.

Reply:
Deleted the proposal in the comments. 
Living with strings for Date.
Added JSDoc comments for users.